src/ORM/Query.php

```diff
@@ -287,7 +287,7 @@ public function eagerLoader(EagerLoader $instance = null)
      */
     public function contain($associations = null, $override = false)
     {
-        if (empty($associations) && $override) {
+        if ($override) {
             $this->_eagerLoader = null;
```

Comment:
The same eagerloader instance is used for storing `matching`. Should everything be cleared out with contain override? If not, we need another method to clear `matching` and the `joinWith`